src/main/java/com/google/cloud/hadoop/io/bigquery/hive/HiveBigQueryMetaHook.java

```diff
@@ -0,0 +1,92 @@
+package com.google.cloud.hadoop.io.bigquery.hive;
+
+import com.google.common.base.Strings;
+import org.apache.hadoop.hive.metastore.HiveMetaHook;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *  Class {@link HiveBigQueryMetaHook} can be used to validate and perform different actions during creation and
+ *  dropping of Hive Table.
+ */
+public class HiveBigQueryMetaHook implements HiveMetaHook {
+
+    private static Logger LOG = LoggerFactory.getLogger(HiveBigQueryMetaHook.class);
+
+    /**
+     *  Performs required validations prior to creating the table
+     *
+     * @param table Represents hive table object
+     * @throws MetaException
+     */
+    @Override
+    public void preCreateTable(Table table) throws MetaException {
+
+
+        //Check all mandatory table properties
+        for(String property : HiveBigQueryConstants.MANDATORY_TABLE_PROPERTIES) {
+            if (Strings.isNullOrEmpty(table.getParameters().get(property))) {
+                throw new MetaException(property + " table property cannot be empty.");
+            }
+        }
+
+        //Check compatibility with BigQuery features
+        //TODO: accept DATE column 1 level partitioning
+        if(table.getPartitionKeysSize() > 0) {
+                throw new MetaException("Creation of Partition table is not supported.");
+        }
+
+        if(table.getSd().getBucketColsSize() > 0) {
+            throw new MetaException("Creation of bucketed table is not supported");
+        }
+
+        if(table.getTableType().equals("EXTERNAL_TABLE")) {
+            throw new MetaException("Creation of External table is not supported.");
+        }
+
+         /*
+        if(!Strings.isNullOrEmpty(table.getSd().getLocation())) {
```

Comment:
Why removing this location check?

Reply:
Location is normally specified during creation of external tables. Since external table check has been added I commented the location check.  Since there is no harm in Checking I uncommented the code

Reply:
location can be used with managed table as well. so checking is necessary. external table check is not necessary.